### PR TITLE
Goptions.declare_* functions return unit instead of a write_function

### PR DESCRIPTION
**Pull request by SkySkimmer**
Returning a writer insinuates that it is not exactly the same as the
writer which was passed as argument, but that is incorrect.

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]
OK with this, let's see if it breaks anything.

File: plugins/cc/ccalgo.ml [hunk 1]
```diff
@@ -33,7 +33,7 @@ let print_constr t =
 let debug x =
   if !cc_verbose then Feedback.msg_debug (x ())
 
-let _=
+let ()=
```
Comment (vbgl) [inline plugins/cc/ccalgo.ml]: Missing whitespace?

File: plugins/firstorder/g_ground.mlg [hunk 2]
```diff
@@ -47,7 +47,7 @@ let _=
     declare_int_option gdopt
 
 
-let _=
+let ()=
```
Comment (vbgl) [inline plugins/firstorder/g_ground.mlg]: Whitespaces.

File: plugins/ssr/ssrfwd.ml [hunk 3]
```diff
@@ -67,13 +67,13 @@ open Ssripats
...
 
 let _ =
-  Goptions.declare_bool_option
-    { Goptions.optname  = "have type classes";
-      Goptions.optkey   = ["SsrHave";"NoTCResolution"];
-      Goptions.optread  = (fun _ -> !ssrhaveNOtcresolution);
-      Goptions.optdepr  = false;
-      Goptions.optwrite = (fun b -> ssrhaveNOtcresolution := b);
-    }
+  Goptions.(declare_bool_option
```
Comment (vbgl) [inline plugins/ssr/ssrfwd.ml]: `let () =` instead of `let _ =`

File: pretyping/evarconv.ml [hunk 4]
```diff
@@ -33,14 +33,14 @@ type unify_fun = TransparentState.t ->
   env -> evar_map -> conv_pb -> EConstr.constr -> EConstr.constr -> Evarsolve.unification_result
 
 let debug_unification = ref (false)
-let _ = Goptions.declare_bool_option {
-  Goptions.optdepr = false;
-  Goptions.optname =
+let _ = Goptions.(declare_bool_option {
```
Comment (vbgl) [inline pretyping/evarconv.ml]: `let ()`

**Review by vbgl — APPROVED** [review]

**Comment by ejgallego** [comment]
This seem ready to go; maintainers of this file are @silene and @gares tho @herbelin has also expressed interest in `GOptions`. I can also assign it if the maintainers are busy.

**Comment by ejgallego** [comment]
Will merge tomorrow if maintainers remain silent.

**Comment by herbelin** [comment]
Good to me.

**Pull request merged**